conn.go

```diff
@@ -9,18 +9,36 @@ import (
 	"time"
 )
 
+// DefaultConnTimeout time in seconds to wait for connection to beanstalk server.
+const DefaultConnTimeout = 10
+
+// DefaultReadTimeout time in seconds to wait for response from beanstalk server.
+const DefaultReadTimeout = 10
```

Comment:
Please don't put completely arbitrary deadlines in libraries by default.

10 seconds may seem reasonable most of the time, but if I have a batch process failing operations because for some reason the request takes 10.5s, I'm going to be very angry at someone if that gets me paged.

A connection timeout is reasonable to have, but I'm not a fan of default deadlines because they turn performance problems to debug at my leisure into hard-failures that wake someone up at night (or give them an error-page).

Reply:
I originally provided the ability to set a custom timeout by @JensRantil wasn't keen on changing the interface, so I made the timeouts (what I thought) was a sensible default.

Your comment about the batch/job times taking longer than the arbitrary time is valid, but I dont believe this will be an issue, as I have considered it.

This is because when a job has been received and is being processed by a consumer, there is no timeout in effect because the package is not 'reading' (or waiting for a response) from the beanstalk server - so the read timeout doesn't apply.

Where I had to be careful when implementing this is the reserve request, which can wait an arbitrary time specified by the client to wait for a job before getting up.

However I have handled this by setting the read timeout for that request to the wait time + 10s. Here https://github.com/beanstalkd/go-beanstalk/pull/26/files#diff-a2ddca1600f34abb91244424f8431c2cR174

Hope this allays your concerns.